test/unit/org/apache/cassandra/db/WriteOptionsTest.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db;
+
+import java.nio.ByteBuffer;
+import java.util.Set;
+
+import com.google.common.collect.Sets;
+import org.junit.Test;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.cql3.CQLTester;
+
+import static org.apache.cassandra.db.WriteOptions.DEFAULT;
+import static org.apache.cassandra.db.WriteOptions.DEFAULT_WITHOUT_COMMITLOG;
+import static org.apache.cassandra.db.WriteOptions.FOR_BATCH_REPLAY;
+import static org.apache.cassandra.db.WriteOptions.FOR_BOOTSTRAP_STREAMING;
+import static org.apache.cassandra.db.WriteOptions.FOR_HINT_REPLAY;
+import static org.apache.cassandra.db.WriteOptions.FOR_PAXOS_COMMIT;
+import static org.apache.cassandra.db.WriteOptions.FOR_STREAMING;
+import static org.apache.cassandra.db.WriteOptions.FOR_VIEW_BUILD;
+import static org.apache.cassandra.db.WriteOptions.SKIP_INDEXES_AND_COMMITLOG;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class WriteOptionsTest extends CQLTester
+{
+    @Test
+    public void testShouldWriteCommitLog() throws Throwable
+    {
+        String DURABLE_KEYSPACE = "durable_ks";
+        String NON_DURABLE_KEYSPACE = "non_durable_ks";
+
+        Set<WriteOptions> WRITE_COMMIT_LOG_TRUE = Sets.newHashSet(FOR_BOOTSTRAP_STREAMING,
+                FOR_STREAMING,
+                FOR_PAXOS_COMMIT,
+                FOR_VIEW_BUILD);
+        Set<WriteOptions> WRITE_COMMIT_LOG_AUTO = Sets.newHashSet(DEFAULT,
+                FOR_BATCH_REPLAY,
+                FOR_HINT_REPLAY);
+
+        schemaChange(String.format("CREATE KEYSPACE IF NOT EXISTS %s WITH replication = {'class': 'SimpleStrategy', 'replication_factor': '1'} AND durable_writes = true", DURABLE_KEYSPACE));
```

Comment:
do you mind adding one more keyspace without `durable_writes` to verify the default behavior?